wire-ios/Wire-iOS/Sources/UserInterface/PermissionDeniedHint/IsNotificationAuthorizationStatusDeniedUseCase/ShouldPresentNotificationPermissionHintUseCase.swift

```diff
@@ -0,0 +1,57 @@
+//
+// Wire
+// Copyright (C) 2024 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import UserNotifications
+import WireSystem
+import WireUtilities
+
+struct ShouldPresentNotificationPermissionHintUseCase<DateProvider>: ShouldPresentNotificationPermissionHintUseCaseProtocol
+where DateProvider: CurrentDateProviding {
+
+    var currentDateProvider: DateProvider
+    var userDefaults: UserDefaults
+    var userNotificationCenter: UserNotificationCenterAbstraction
+
+    func invoke() async -> Bool {
+
+        let lastPresentationDate = userDefaults.value(for: .lastTimeNotificationPermissionHintWasShown)
+        if let lastPresentationDate, lastPresentationDate < currentDateProvider.now.addingTimeInterval(-24 * 3600) {
```

Comment:
A constant would help to understand:
```
private enum Constant {
    static let oneDay: Int = 24 * 60 * 60
}
```

Reply:
We actually have constants already, fixed.